src/content/docs/blog/2025/07/11/Development update: os-tools/index.md

```diff
@@ -0,0 +1,184 @@
+
+Given that boulder also needs to blit files when it creates buildroots, the code has also had a positive impact on reducing package build times. This will be more evident on larger package builds and will have a cumulative impact, the more package work you end up doing.
+
+### Moss: Sync available before installed packages
+
+As we were testing the upgrade path from our old `packages.aerynos.com/volatile` repository to our new, CDN-backed `cdn.aerynos.dev/unstable` repository, we ran into some unexpected small niggles related to how packages are resolved.
+
+While tarkah's [fix](https://github.com/AerynOS/os-tools/pull/514) to Moss was relatively small in terms of code, it served to ensure that updates would install properly on the first go when syncing to the new repo.
+
+Consequently, we synced this bug-fix to the Moss version in the old repository to ensure that users will be able to seamlessly upgrade to the new rolling `unstable` repository.
+
+
+### Moss: Add index output directory option
+
+As we were preparing the process for sync'ing the packages in our `volatile` build-server repository to our new downstream rolling `unstable` repository on our public-facing server, we ran into an issue with the existing `moss index` code path.
+
+Before this issue was fixed, the `stone.index` file would be unconditionally written next to the actual package .stone files. This was useful when indexing local repos, but not as useful when indexing actual stone [pool/](https://cdn.aerynos.dev/unstable/pool/) directories and sub-directories.
+
+In the end, this was another small [feature](https://github.com/AerynOS/os-tools/pull/523) with somewhat large consequences, in that this enabled us to do the actual manual indexing in a way that is identical to how our infrastructure organizes things when indexing.
+
+This in turns made it possible to make sure that the new rolling `unstable` repo presents with the same URI "pattern" as our volatile build repo:
+
+    ❯ moss lr
+     - oldrepo = https://packages.aerynos.com/volatile/x86_64/stone.index [0]
+     - unstable = https://cdn.aerynos.dev/unstable/x86_64/stone.index [5]
+     - volatile = https://build.aerynos.dev/volatile/x86_64/stone.index [10]
+     - local = file:///home/ermo/.cache/local_repo/x86_64/stone.index [100]
+
+
+### Boulder: Fix up phase timing in end-of-build report
+
+When Boulder successfully completes a package build, it emits a report detailing how long each phase of the build process took.
+
+ermo noticed that the output was wrong when the time exceeded an hour. For example 76 minutes would be formatted as a rather silly-looking `1h76m` instead of `1h11m`. He fixed this in the following [commit](https://github.com/AerynOS/os-tools/pull/502).
```

Comment:
```suggestion
ermo noticed that the output was wrong when the time exceeded an hour. For example 76 minutes would be formatted as a rather silly-looking `1h76m` instead of `1h16m`. He fixed this in the following [commit](https://github.com/AerynOS/os-tools/pull/502).
```